// react-logos/lib/GuruDigital.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Hypers = _react2.default.createClass({
    displayName: 'Hypers',

    propTypes: {
        width: _react2.default.PropTypes.number,
        backgroundColor: _react2.default.PropTypes.string,
        color: _react2.default.PropTypes.string
    },
    getDefaultProps: function getDefaultProps() {
        return {
            width: 90,
            backgroundColor: '#00b12b',
            color: '#fff'
        };
    },

    render: function render() {
        var _props = this.props;
        var width = _props.width;
        var color = _props.color;
        var backgroundColor = _props.backgroundColor;

        var gridWidth = width / 9;
        var polygonPoints = [gridWidth * 2 + ',' + gridWidth * 2, gridWidth * 7 + ',' + gridWidth * 2, gridWidth * 7 + ',' + gridWidth * 3, gridWidth * 3 + ',' + gridWidth * 3, gridWidth * 3 + ',' + gridWidth * 6, gridWidth * 6 + ',' + gridWidth * 6, gridWidth * 6 + ',' + gridWidth * 5, gridWidth * 4 + ',' + gridWidth * 5, gridWidth * 4 + ',' + gridWidth * 4, gridWidth * 7 + ',' + gridWidth * 4, gridWidth * 7 + ',' + gridWidth * 7, gridWidth * 2 + ',' + gridWidth * 7];

        var rect = _react2.default.createElement('rect', { x: 0, y: 0, width: width, height: width, fill: backgroundColor, strokeWidth: '0' });
        var polygon = _react2.default.createElement('polygon', { points: polygonPoints.join(' '), fill: color, strokeWidth: '0' });

        return _react2.default.createElement(
            'svg',
            { width: width, height: width },
            rect,
            polygon
        );
    }
});

exports.default = Hypers;
