// react-logos/lib/Hypers.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Hypers = _react2.default.createClass({
    displayName: 'Hypers',

    propTypes: {
        width: _react2.default.PropTypes.number,
        color: _react2.default.PropTypes.string,
        backgroundColor: _react2.default.PropTypes.string
    },
    getDefaultProps: function getDefaultProps() {
        return {
            width: 90,
            color: '#00c3ee',
            backgroundColor: 'transparent'
        };
    },

    render: function render() {
        var _props = this.props;
        var width = _props.width;
        var color = _props.color;
        var backgroundColor = _props.backgroundColor;

        var gridWidth = width / 9;

        var polygonPoints = [gridWidth * 2 + ',' + gridWidth * 2, gridWidth * 3 + ',' + gridWidth * 2, gridWidth * 3 + ',' + gridWidth * 4, gridWidth * 7 + ',' + gridWidth * 4, gridWidth * 7 + ',' + gridWidth * 7, gridWidth * 6 + ',' + gridWidth * 7, gridWidth * 6 + ',' + gridWidth * 5, gridWidth * 3 + ',' + gridWidth * 5, gridWidth * 3 + ',' + gridWidth * 7, gridWidth * 2 + ',' + gridWidth * 7];

        var rect = _react2.default.createElement('rect', { x: gridWidth / 2, y: gridWidth / 2, width: gridWidth * 8, height: gridWidth * 8, stroke: color, fill: backgroundColor, strokeWidth: gridWidth });
        var dot = _react2.default.createElement('rect', { x: gridWidth * 6, y: gridWidth * 2, width: gridWidth, height: gridWidth, fill: color, strokeWidth: '0' });
        var polygon = _react2.default.createElement('polygon', { points: polygonPoints.join(' '), fill: color, strokeWidth: '0' });

        return _react2.default.createElement(
            'svg',
            { width: width, height: width },
            rect,
            dot,
            polygon
        );
    }
});

exports.default = Hypers;
